### PR TITLE
Updated User Config Handling

### LINKED ISSUE #723: Session user name disappear on the old device when link a new device
**Code of conduct**

- [x] I have read and agree to adhere to the [Code of Conduct](https://github.com/oxen-io/session-ios/blob/master/CODE_OF_CONDUCT.md).

**Describe the bug**

This might be a regression, not sure when it starts. Reproducible on both Session iOS 2.2.0 and Session iOS 2.2.1.

**To reproduce**

Old device: Session Desktop / Session iOS
New device: Session iOS

Steps:

1. create a fresh new Session ID on an old device with the user name "Test"
2. "Link a device" from the new device to the old device

Expect result:
Both old device and new device has the same user name "Test"

Actual result:
Old device has no user name anymore, not sure if it's an empty string or a NULL pointer.
New device has the correct user name "Test"
Edit: after syncing for a while, new device lost its user name as well.

It seems impossible to edit the user name on the old device anymore, the user name input box on the old device is non-editable after linking the new device.

Session user name

You can also tell from the Avatar placeholder, it's no longer an abbreviate of user name, it becomes completely empty.

If applicable, add screenshots or logs to help explain your problem.

**Smartphone (please complete the following information):**

 - Device: iPhone SE, iPad
 - OS: iOS 15.6
 - Version of Session: 2.2.0 / 2.2.1

**Additional context**

We need better test cases to cover this kind of weird edge case :D

### PR DESCRIPTION
**Pull request by mpretty-cyro**
**Changes:**
- Replaced PromiseKit usage with Combine
- Replaced AFNetworking with standard URLSession
- Integrated the libSession-util and updated the user config syncing to use it
- Minimised usages of CryptoSwift (Using CryptoKit where possible) and Curve25519
- Cleaned up some of the SessionTableViewModel/SessionTableViewController handling
- Cleaned up the profile updating logic
- Removed the iOS-specific concurrent dual snode '/store' behaviour
- Tweaked the code to only update the contact permission flags if they actually change (avoid unneeded db changes in case something is observing them)
- Updated the onboarding to attempt to retrieve the current user profile config and skip display name collection if it already exists
- Updated the logic to get the snode pool and build paths immediately on launch even if the user hasn't been created yet (faster onboarding)
- Fixed an issue where the pollers could end up deadlocking the main thread if too many tried to start concurrently
- Fixed a crash which could occur when downloading new avatars
- Fixed an issue where open group info could incorrectly get removed
- Fixed an edge case where an OpenGroup might not download it's image
- Fixed a minor layout issue with the seed reminder view
- Fixed an issue where "shadow" conversations would be left in the database by opening a thread and never sending a message
- Fixed a bug where duplicate members could be added to legacy groups
- Updated the profile picture modal
- Standardised the ProfilePictureView sizes
- Moved the ProfilePictureView into SessionUIKit
- Fixed an issue where 'CurrentAppContext().isMainAppAndActive' wasn't called on the main thread Updated the ProfilePictureView to have the updated icon UI
- Fixed a bug where reacting to an incoming attachment message in a one-to-one conversation would result in the attachment being re-uploaded
- Fixed a bug where concurrent queues in the JobRunner wouldn't start newly added jobs until current jobs had finished executing

Fixes #723

### CODE REVIEW

**Review by RyanRory — APPROVED** [review]

File: SessionMessagingKit/SessionUtil/Config Handling/SessionUtil+Contacts.swift [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+                                (!data.contact.didApproveMe || contact.didApproveMe == data.contact.didApproveMe ? nil :
+                                    Contact.Columns.didApproveMe.set(to: true)
+                                )
+                            ].compactMap { $0 }
+                        )
+                }
+                
+                /// If the contact's `hidden` flag doesn't match the visibility of their conversation then create/delete the
+                /// associated contact conversation accordingly
+                let threadInfo: PriorityVisibilityInfo? = try? SessionThread
```
Comment (RyanRory) [inline SessionMessagingKit/SessionUtil/Config Handling/SessionUtil+Contacts.swift:160]: There should be a `.filter(id: sessionId)` I reckon.

Comment (mpretty-cyro) [reply SessionMessagingKit/SessionUtil/Config Handling/SessionUtil+Contacts.swift:160]: Yep good catch, have updated

Comment (mpretty-cyro) [reply SessionMessagingKit/SessionUtil/Config Handling/SessionUtil+Contacts.swift:160]: Surprised this hasn't caused more bugs actually 🤔

Comment (RyanRory) [reply SessionMessagingKit/SessionUtil/Config Handling/SessionUtil+Contacts.swift:160]: Yeah, it's strange that this didn't cause more bugs.

File: Session/Conversations/Message Cells/VisibleMessageCell.swift [hunk 2]
```diff
@@ -22,7 +22,6 @@ final class VisibleMessageCell: MessageCell, TappableLabelDelegate {
     private lazy var authorLabelTopConstraint = authorLabel.pin(.top, to: .top, of: self)
     private lazy var authorLabelHeightConstraint = authorLabel.set(.height, to: 0)
     private lazy var profilePictureViewLeadingConstraint = profilePictureView.pin(.leading, to: .leading, of: self, withInset: VisibleMessageCell.groupThreadHSpacing)
-    private lazy var profilePictureViewWidthConstraint = profilePictureView.set(.width, to: Values.verySmallProfilePictureSize)
```
Comment (RyanRory) [inline Session/Conversations/Message Cells/VisibleMessageCell.swift:25]: Remove this constraint will end up with a UI bug for incoming messages in 1-1 convos. The leading constraint of content view is based on the trailing of profile picture view, if there is no width of profile picture view, the leading constraint of content view will not work.

Comment (mpretty-cyro) [reply Session/Conversations/Message Cells/VisibleMessageCell.swift:25]: Ah yep I see it - sorry this change hasn't gone through QA yet but should as User Config is getting tested, I'll push up a fix now (I'll have the `ProfilePictureView` update it's constraints when `isHidden` is updated rather than adding this back since it's managing it's own size constraints now)
